src/openms/source/ANALYSIS/MRM/ReactionMonitoringTransition.cpp

```diff
@@ -205,6 +205,22 @@ namespace OpenMS
     }
   }
 
+  int ReactionMonitoringTransition::getProductChargeState() const
+  { 
+    try
+    { 
+      if (!product_.getChargeState() > 0)
```

Comment:
here, the code is a bit confusing. did you mean 
```
if (!(product_.getChargeState() > 0))
```
note the brackets (operator `!` takes precedence http://en.cppreference.com/w/cpp/language/operator_precedence ).  or easier
```
if (product_.getChargeState() <= 0)
```